python/iot3/src/iot3/core/__init__.py

```diff
@@ -0,0 +1,211 @@
+    :param message_callback: The function to call when a message is
+                             received; when called, the first argument
+                             will be callback_data, the second will be
+                             the topic the emsage was received on, and
+                             the third and last argument will be the
+                             message payload.
+    :param callback_data: The data to pass as first argument to
+                          message_callback(), above.
+    """
+    global _core
+
+    if _core is not None:
+        raise RuntimeError("IoT3 Core SDK already intialised.")
+
+    o_kwargs = dict()
+
+    for auth in _otel.Auth:
+        if config["otel"]["auth"] == auth.value:
+            o_kwargs["auth"] = auth
+            if auth != otel.Auth.NONE:
+                o_kwargs["username"] = config["otel"]["username"]
+                o_kwargs["password"] = config["otel"]["password"]
+            break
+    else:
+        raise ValueError(f"unknown authentication {config['otel']['auth']}")
+
+    for comp in _otel.Compression:
+        if config["otel"]["compression"] == comp.value:
+            o_kwargs["compression"] = comp
+            break
+    else:
+        raise ValueError(f"unknown compression {config['otel']['compression']}")
+
+    o = _otel.Otel(
+        service_name=config["otel"]["service_name"],
+        endpoint=config["otel"]["endpoint"],
+        batch_period=config["otel"]["batch_period"],
+        max_backlog=config["otel"]["max_backlog"],
+        **o_kwargs,
+    )
+    o.start()
+
+    # Wrap the callback to avoid leaking telemetry into the simple API
+    def _msg_cb(*, data, topic, payload, **_kwargs):
+        message_callback(data, topic, payload)
+
+    m = _mqtt.MqttClient(
+        client_id=config["mqtt"]["client-id"],
+        host=config["mqtt"]["host"],
+        port=config["mqtt"]["port"],
+        username=config["mqtt"]["username"],
+        password=config["mqtt"]["password"],
+        msg_cb=_msg_cb if message_callback else None,
+        msg_cb_data=callback_data,
+        span_ctxmgr_cb=o.span,
+    )
+    m.start()
+
+    _core = dict([("otel", o), ("mqtt", m)])
+
+
+def stop():
+    """Stop the Iot3 Core SDK."""
+    global _core
+
+    if _core is None:
+        raise RuntimeError("IoT3 Core SDK not initialised.")
+
+    _core["mqtt"].stop()
+    _core["otel"].stop()
+    _core = None
+
+
+def is_ready() -> bool:
+    """Checkxs whether the IoT3 Core SDK is ready.
+
+    The IoT3 Core SDK is ready when the MQTT client is connected to
+    the MQTT broker. Messages sent when the SDK is not ready, are
+    lost without notice.
+    """
+    global _core
+
+    if _core is None:
+        raise RuntimeError("IoT3 Core SDK not initialised.")
+
+    return _core["mqtt"].is_ready()
+
+
+def wait_for_ready():
+    """Wait until the IoT2 Core SDK is ready.
+
+    Beware that this may take an undeterminate amount of time; in
```

Comment:
undetermined

Reply:
There was indeed a typo. It's in fact spelled [_indeterminate_](https://www.merriam-webster.com/dictionary/indeterminate). :-)